src/main/java/org/opensearch/security/OpenSearchSecurityPlugin.java

```diff
@@ -1806,7 +1805,9 @@ public void onNodeStarted(DiscoveryNode localNode) {
         if (!SSLConfig.isSslOnlyMode() && !client && !disabled) {
             cr.initOnNodeStart();
```

Comment:
Why can't `si.setLocalNode(localNode);` be called in this block?

It looks like the SecurityInterceptor component is created if this condition evaluates to true based on these lines: https://github.com/opensearch-project/security/blob/main/src/main/java/org/opensearch/security/OpenSearchSecurityPlugin.java#L919-L943

Reply:
I'm not sure about when will client and disabled flags be set to true or false.

From the link that you mentioned, SecurityInterceptor will not be initialized if client or disabled flags are set to true.

but the condition in the present case explicitly asks `&& !client && !disabled` to be true  and so SecurityInterceptor localNode will not be initialized if either of client or disabled is set to true.

Reply:
We should use an AtomicReference to retrieve/capture this value when it comes in onNodeStarted(), I've pushed a change that captures this.  It still returns null if it wasn't set, but that is OK since the fallback handles this.